### ISSUE TITLE
Install Failed in ASP.NET Core 3.1 project

### ISSUE THREAD

**Issue by codeease1** [statiqdev/Statiq.Framework#87]
Hi:

I am not sure whether the current Statiq.Framework can be used in an ASP.NET Core 3.1 MVC project?

I created an empty ASP.NET Core 3.1 MVC project in Visual Studio 2019, and then used Nuget Manager to install Statiq.App, Statiq.Common and Statiq.Markdown successfully. but later I found that I could not add any new controller class (right click the Controller folder and add new controller);

I realized it could be caused by the Statiq stuff. then I uninstalled Statiq.App, Statiq.Common and Statiq.Markdown,  then I could add a new controller class;

After I added the new controller class, I tried to install the 3 Statiq libs again, but I got failed messages:

_Error	NU1107	Version conflict detected for Microsoft.CodeAnalysis.Common. Install/reference Microsoft.CodeAnalysis.Common 3.4.0 directly to project TryStatiq2 to resolve this issue. 
 TryStatiq2 -> Statiq.App 1.0.0-alpha.27 -> Statiq.Core 1.0.0-alpha.27 -> Microsoft.CodeAnalysis.CSharp 3.4.0 -> Microsoft.CodeAnalysis.Common (= 3.4.0) 
 TryStatiq2 -> Microsoft.VisualStudio.Web.CodeGeneration.Design 3.1.1 -> Microsoft.VisualStudio.Web.CodeGenerators.Mvc 3.1.1 -> Microsoft.VisualStudio.Web.CodeGeneration 3.1.1 -> Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore 3.1.1 -> Microsoft.VisualStudio.Web.CodeGeneration.Core 3.1.1 -> Microsoft.VisualStudio.Web.CodeGeneration.Templating 3.1.1 -> Microsoft.VisualStudio.Web.CodeGeneration.Utils 3.1.1 -> Microsoft.CodeAnalysis.CSharp.Workspaces 3.3.1 -> Microsoft.CodeAnalysis.Common (= 3.3.1)_.

**Comment by codeease1** [comment]
Update:
Today I manually installed Microsoft.CodeAnalysis.Common 3.4.0 in Nuget Manager, the issue was resolved.

**Comment by daveaglick** [comment]
Great! Glad that worked. I suspect this may be an ongoing "problem". Given how much Microsoft.CodeAnalysis is used in tooling like Visual Studio, any library that also uses it will likely be out of sync version-wise from time-to-time. When that happens, manually installing the appropriate version helps remove the ambiguity. The alternative is for libraries to try and chase whatever version of Microsoft.CodeAnalysis is currently used by the tooling, but that's a loosing proposition since doing so is time consuming and probably won't keep up anyway.

**Comment by codeease1** [comment]
Sorry, what I meant is that the issue is still there.

If I installed the Statiq lib in an ASP.NET Core 3.1 mvc project in Visual Studio 2019, whatever I manually update the Microsoft.CodeAnalysis.Common versions, I still could not add Controller or Views using Visual Studio's Scaffolding.  What I resolved issue is if I added a new Controller first then I could not install Statiq lib with manually updating  Microsoft.CodeAnalysis.Common.  but if I installed Statiq Lib (whenever), I no longer be able to add new Controller and Views.
